CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+them of your awesome contributions.
+Based on the comments posted by the reviewers, you may have to revisit your patches.
+
+
+#### How to efficiently contribute ?
+
+We encourage all developers to:
+
+* review and follow the style guidelines described
+[on the wiki](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Style_Guide#Commits).
+
+* consider potential backward compatibility breakage and discuss these on the
+  mailing list. For example, update of ITK, Python, Qt or VTK version, change to
+  core functionality, should be carefully reviewed and integrated. Ideally, several
+  developers would test that the changes don't break extensions.
+
+#### How to integrate a PR ?
+
+Getting your contributions integrated is relatively straightforward, here
+is the checklist:
+
+* All tests pass
+* Consensus is reached. This usually means that at least two reviewers approved
+  the changes (or added a `LGTM` comment) and at least one business day passed
+  without anyone objecting. `LGTM` is an acronym for _Looks Good to Me_.
+* To accommodate developers explicitly asking for more time to test the
+  proposed changes, integration time can be delayed by few more days.
+
+Next, there are two scenarios:
+* You do NOT have push access: A Slicer core developer will integrate your PR. If
+  you would like to speed up the integration, do not hesitate to send a note on
+  the mailing list.
+* You have push access: Follow [Integrating topic from external contributor](https://www.slicer.org/wiki/Slicer:git-svn#Integrating_topic_from_external_contributor)
+  instructions on the wiki.
+
+
+#### Automatic testing of pull requests
+
+Every pull request is tested automatically using CircleCI each time you push a
+commit to it. The Github UI will restrict users from merging pull requests until
+the CI build has returned with a successful result indicating that all tests have
+passed.
+
+The testing infrastructure is described in details in the
+[3D Slicer Improves Testing for Pull Requests Using Docker and CircleCI](https://blog.kitware.com/3d-slicer-improves-testing-for-pull-requests-using-docker-and-circleci/)
+blog post.
+
+
+#### Nightly tests
+
+After changes are integrated, every evening at 10pm EST (3am UTC), Slicer build bots (aka factories)
+will build, test and package Slicer application and all its extensions on Linux, MacOSX
+and Windows. Results are published daily on [CDash](http://slicer.cdash.org/index.php?project=Slicer4)
+and developers introducing changes introducing build or test failures are notified by
+email.
+
+
+#### Decision-making process
+
+1. Given the topic of interest, initiate discussion on the [mailing list](http://massmail.spl.harvard.edu/mailman/listinfo/slicer-devel).
+
+2. Identify a small circle of community members that are interested to study the
+   topic in more depth.
+
+3. Take the discussion off the general list, work on the analysis of options and
```

Comment:
👎  I don't think there is any reason to take code discussion off-list. It's not like the mailing list is very busy, and archived discussions can be helpful to future code explorers.